doc/docs/master/deploy-manage/deploy/envoy.md

```diff
@@ -0,0 +1,487 @@
+
+Before any deployment in production, we recommend reading the following recommendations section and [set up your production infrastructure](#deploy-pachyderm-in-production-with-envoy). 
+
+Alternatively, you can skip those infrastructure prerequisites and make a [quick cloud installation](#quick-cloud-deployment-with-envoy) or jump to our [local deployment](#deploy-pachyderm-locally-with-envoy) section for the first encounter with Pachyderm.
+
+## Pachyderm General Infrastructure Recommendations
+
+For production deployments, we recommend that you:
+
+* **Provision a TCP load balancer** for all HTTP/HTTPS, gRPC/gRPCs, aws s3, /dex incoming traffic.
+The TCP load balancer (load balanced at L4 of the OSI model) will have port `80/443` forwarding to the `pachyderm-proxy` service entry point. Please take a look at the diagram above.
+        
+    When Envoy is enabled with `type:LoadBalancer` (see the snippet of values.yaml enabling the proxy), Pachyderm creates a `pachyderm-proxy` service allowing your cloud platform (AWS, GKE...) to **provision a TCP Load Balancer automatically**.
+        
+    To provision this external load balancer automatically (if supported) and attach any Load Balancer configuration information to the metadata of your servic, add the appropriate `annotations` in the `proxy.service` of your values.yaml (see below):
+
+    ```yaml
+    proxy:
+      enabled: true
+      service:
+        type: LoadBalancer
+        annotations: {see examples below}
+    ```
+
+
+    === "Example on AWS EKS"
+        In the following example, we deploy an NLB on AWS EKS:
+
+        ``` yaml
+        proxy:
+          enabled: true
+          service:
+            type: LoadBalancer
+            annotations: {see examples below}
+              service.beta.kubernetes.io/aws-load-balancer-type: "external"
+              service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: "ip"
+              service.beta.kubernetes.io/aws-load-balancer-scheme: "internal"
+              service.beta.kubernetes.io/aws-load-balancer-subnets: "subnet-aaaaa,subnet-bbbbb,subnet-ccccc"
+
+        ```
+
+    === "Example on GCP GKE"
+        In the following example, we pre-created a static IP by running `gcloud compute addresses create ADDRESS_NAME --global --ip-version IPV4`, then passed this external IP to the values.yaml as follow:
+
+        ``` yaml
+        proxy:
+          enabled: true
+          service:
+            type: LoadBalancer
+            loadBalancerIP: ${ADDRESS_NAME}
+        ```
+    === "Example on Azure AKS"
+        This example is identical to the example on Google GKE.
+
+        ``` yaml
+          proxy:
+            enabled: true
+            service:
+              type: LoadBalancer
+              loadBalancerIP: ${ADDRESS_NAME}
+        ```
+
+* **Use a secure connection**
```

Comment:
This won't work with the proxy.  pachd and console TLS has to be enabled, and then the secret has to be in `envoy-tls`, and `proxy.tls.enabled` in the values will need to be true.  (This code isn't merged yet, however.)

Reply:
I removed the TLS section for now with a Coming soon mention.